app/services/onlinetesting/OnlineTestService.scala

```diff
@@ -129,3 +129,16 @@ trait OnlineTestService extends EventSink  {
     }
   }
 }
+
+trait TimeExtension {
+  val dateTimeFactory: DateTimeFactory
+
+  def extendTime(alreadyExpired: Boolean, previousExpirationDate: DateTime) = { extraDays: Int =>
```

Comment:
I personally would add the return type to this but it's something that can be done later